### ISSUE TITLE
[FEATURE] consider audiobook support

### ISSUE THREAD

**Issue by Guscal** [stumpapp/stump#177]
i feel there is a lack of an all in one solution maybe stumps ist that?  maybe considered for the long run in a much later update

**Comment by aaronleopold** [comment]
This is definitely something I've considered a few times since starting Stump, and coincidentally something that was recently chatted about on the Discord server. In general, before I would seriously consider the effort of adding audio support of any kind, I would want to reach some form of stability in what Stump currently can/aims to do with digital books. Adding too much before that point would just make development and maintenance harder IMO, and I don't want to pollute Stump with _more_ unfinished features just yet 🙂  There are also some aspects that I would want to very clearly define before starting, since the work involved would be VERY large and I don't want to have to plan or get stuck on a question in the middle of integration more than is necessary. For example: how would the UI change, would we want a clear separation of media types or should audio and digital books be allowed in a single library, how should the DB model(s) change to balance minimizing breaking changes while facilitating an easier implementation, are there good audio metadata parsing libraries available in Rust (I don't feel like creating one myself on top of all this), etc.

TLDR; in the _long_ run, yes, I'm open to considering it

**Comment by kennywasafk** [comment]
Apologies, couldn't find a more relevant issue thread to ask this, but the latest update on Audiobook support is this https://github.com/stumpapp/stump/issues/279 and it seems to have been about a year back. I just wanted to check in and see what you're planning if anything since I really like the project!

**Comment by aaronleopold** [comment]
No apologies needed, this is the perfect spot to ask about audiobook support! This effort hasn’t been started yet, I have personal notes about how it might be developed, ideas for database and API changes, but that’s about it.

It likely won’t be started until the more core features planned for `0.1.0` are implemented, or it could be potentially started earlier if there are more active contributors to the project. But as it currently stands, it would be an enormous effort with fundamentally breaking changes to everyone’s libraries that would require the bulk of my time and focus. Ideally Stump would be at a point where I could comfortably have a feature freeze of sorts, keeping it in maintenance mode until all the audiobook changes are sorted. 

I want to note that the biggest aspect wrt breaking changes is the request to allow mixing audiobooks and digital books in the same library. While this enables things like alternative editions and read-and-listen features, it’s huge. It would be way less effort to just say the audiobooks are a separate library type. But, from what I’ve gathered, folks seem to generally prefer the former. Happy to have a discussion here if that isn't the case. 

Also I enjoy audiobooks more and more lately, so it’s definitely something I want to be added at some point down the road. I just only have so much spare time to do all the things 😅

**Comment by Kernald** [comment]
> I want to note that the biggest aspect wrt breaking changes is the request to allow mixing audiobooks and digital books in the same library. While this enables things like alternative editions and read-and-listen features, it’s huge. It would be way less effort to just say the audiobooks are a separate library type. But, from what I’ve gathered, folks seem to generally prefer the former. Happy to have a discussion here if that isn't the case.

I wonder if there's an opportunity to get things moving here, by adding support for audiobooks separately first, which could likely be done without having to put core things on hold, and then once this is in a good place, merged as a single library to enable whispersync-like, alternative editions... features?

**Comment by aaronleopold** [comment]
> I wonder if there's an opportunity to get things moving here, by adding support for audiobooks separately first, which could likely be done without having to put core things on hold

It isn't necessarily that core things would require being put on hold as much as it is I can only feasibly work on one larger-scale chunk of work at a time when considering my already limited time as it is (full time job, etc). In a way, I have already had to put core things on hold to knock out the [massive rewrite](https://github.com/stumpapp/stump/discussions/634), though eventually I did work on feature development to refresh my energy to _continue_ with the rewrite (which is basically done and in testing at this point).

There is always opportunity to get things moving, it is an open source project and anyone is free to contribute their time and energy to expedite the things they want to see. Historically, though, Stump does not have a large contributor list.
